packages/gatsby-source-medium/src/gatsby-node.js

```diff
@@ -0,0 +1,52 @@
+const axios = require(`axios`)
+const crypto = require(`crypto`)
+
+const fetch = username => {
+  const url = `https://medium.com/${username}/latest?format=json`
+  return axios.get(url)
+}
+
+const prefix = `])}while(1);</x>`
+
+const strip = payload => payload.replace(prefix, ``)
+
+exports.sourceNodes = async ({ boundActionCreators }, { username }) => {
+  const { createNode } = boundActionCreators
+
+  try {
+    const result = await fetch(username)
+    const json = JSON.parse(strip(result.data))
+
+    const { posts } = json.payload
+    const collectionKeys = Object.keys(json.payload.references.Collection)
+    const userKeys = Object.keys(json.payload.references.User)
+
+    const importableResources = [
+      posts,
+      collectionKeys.map(key => json.payload.references.Collection[key]),
+      userKeys.map(key => json.payload.references.User[key]),
+    ]
+
+    const resources = Array.prototype.concat(...importableResources)
+    resources.map(resource => {
+      const digest = crypto.createHash(`md5`).update(JSON.stringify(resource)).digest(`hex`)
+
+      const node = Object.assign(resource, {
```

Comment:
Oh something else that'd be fantastic would be to link posts and users. So that way you could query for users and also then query for their posts and vice versa.

The syntax isn't well-documented but basically to create a link you add a field like `authors___NODE: ["id1", "id2", "id3"]` where `authors` will be the eventual field name (the "___NODE" is stripped) and the array of IDs is the linked user ids who are authors.

Reply:
I'd love to implement this, but honestly, I don't quite get it.

So on resources with type `MediumPost` I'll add a topleve field `author__NODE` (not inside `internal`) which basically has the same content as `resource.creatorId` (Medium User ID) which then magically links to `MediumUser`? I tried that but I'm not sure how to verify whether this works. Should I be able to query:

```
{
  allMediumPost {
    edges {
      node {
        id
      }
      author {}
    }
  }
}
```

Reply:
yeah, the `author___NODE` field should have as its value the "id" of the `MediumUser`. Then you'll be able to query attributes off the `MediumUser` node.

Reply:
Somehow I'm struggling with this. I'm trying the following:

```
const links =
        resource.type === `Post`
          ? {
              author__NODE: resource.creatorId,
            }
          : {}
```

and `links` is also used as a param in the `Object.assign` call. The nodes are added to GraphQL, as they should, but they don't have an `author` field. Any ideas what I'm doing wrong?